### PR TITLE
LPS-169011 check if the url is on a different language to ensure uniqueness

### PR DESCRIPTION
**Pull request by jgarcialr**
Hi Lima Team,

I'm sending this PR to fix a little regression caused by getUniqueURL changes in friendlyURL after LPS-166653

Steps to reproduce:

1. Start master
2. Create a Journal with the default language English. Title A with A as friendlyURL.
3. Add a translation for this article in spanish. Title B with B as friendlyURL
4. Create another journal with the default language English. Title B and B as friendlyURL
5. Add a translation for this article in spanish. Title A with A as friendlyURL.

**Expected behaviour.**
Two different articles should not have the same FriendlyURL

**Current behaviour.**
This different articles have the same FriendlyURL.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -662,8 +662,15 @@ private boolean _hasFriendlyURLEntryWithUrlTitle(
 			friendlyURLEntryLocalizationPersistence.fetchByG_C_L_U(
 				groupId, classNameId, languageId, urlTitle);
 
-		if ((friendlyURLEntryLocalization != null) &&
-			(friendlyURLEntryLocalization.getClassPK() != notClassPK)) {
+		FriendlyURLEntryLocalization friendlyURLEntryLocalizationNotLang =
+			friendlyURLEntryLocalizationPersistence.fetchByG_C_NotL_U_First(
```
Comment (AliciaGarciaGarcia) [inline modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java:666]: What do you think on put the code this way

```
FriendlyURLEntryLocalization friendlyURLEntryLocalization =
	friendlyURLEntryLocalizationPersistence.fetchByG_C_L_U(
		groupId, classNameId, languageId, urlTitle);

if ((friendlyURLEntryLocalization != null) &&
	(friendlyURLEntryLocalization.getClassPK() != notClassPK)) {

	return true;
}

FriendlyURLEntryLocalization friendlyURLEntryLocalizationNotLang =
	friendlyURLEntryLocalizationPersistence.fetchByG_C_NotL_U_First(
		groupId, classNameId, languageId, urlTitle, null);

if ((friendlyURLEntryLocalizationNotLang != null) &&
	(friendlyURLEntryLocalizationNotLang.getClassPK() != notClassPK)) {

	return true;
}
```


double return vs a call to DB (maybe) not needed.

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -662,8 +662,15 @@ private boolean _hasFriendlyURLEntryWithUrlTitle(
 			friendlyURLEntryLocalizationPersistence.fetchByG_C_L_U(
 				groupId, classNameId, languageId, urlTitle);
 
-		if ((friendlyURLEntryLocalization != null) &&
-			(friendlyURLEntryLocalization.getClassPK() != notClassPK)) {
+		FriendlyURLEntryLocalization friendlyURLEntryLocalizationNotLang =
```
Comment (AliciaGarciaGarcia) [inline modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java:665]: What do you think on put the code this way

```
FriendlyURLEntryLocalization friendlyURLEntryLocalization =
	friendlyURLEntryLocalizationPersistence.fetchByG_C_L_U(
		groupId, classNameId, languageId, urlTitle);

if ((friendlyURLEntryLocalization != null) &&
	(friendlyURLEntryLocalization.getClassPK() != notClassPK)) {

	return true;
}

FriendlyURLEntryLocalization friendlyURLEntryLocalizationNotLang =
	friendlyURLEntryLocalizationPersistence.fetchByG_C_NotL_U_First(
		groupId, classNameId, languageId, urlTitle, null);

if ((friendlyURLEntryLocalizationNotLang != null) &&
	(friendlyURLEntryLocalizationNotLang.getClassPK() != notClassPK)) {

	return true;
}
```


double return vs a call to DB (maybe) not needed.

Comment (adolfopa) [reply modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java:665]: I second Alicia's comment. Could you fully spell "Lang" in your variable names too? (e.g. "LanguageId")

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java (hunk 2 above)
Comment (jgarcialr) [reply modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java]: Ok.
I've tested and it works as expected splitting the if's.
I've renamed the var too.

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]
SF problems

But LGTM once solved

**Comment by jgarcialr** [comment]
ci:test:sf

**Comment by jgarcialr** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**